legacy/app/auth/authentication.service.js

```diff
@@ -27,16 +27,25 @@ function (
     // check whether we have initially an valid access_token and assume that, if yes, we are still loggedin
     let loginStatus = false;
     if (!!Session.getSessionDataEntry('accessToken') &&
-        Session.getSessionDataEntry('grantType') === 'password' &&
         !!Session.getSessionDataEntry('userId')
     ) {
         // If the access token is expired
         if (Session.getSessionDataEntry('accessTokenExpires') <= Math.floor(Date.now() / 1000)) {
+            /*
+             * TODO: if there's a refresh token in the session, we should attempt obtaining a new
+             *       access token. BUT should we intercept and defer other requests until we obtain
+             *       the token?
+             */
```

Comment:
> BUT should we intercept and defer other requests until we obtain the token?
I am a bit torn about that... part of me says yes, but another says it may be avoidable complexity. I think we need to reason it a bit more. So, maybe just not for this PR?

In general, I'm wondering if we wouldn't be better off with a fresh approach in the future. I can think of other corner cases that don't seem to be well handled, such as dealing with flaky network connections.

Reply:
Yep, I think so too, time has passes and it looks like there were plans to continue working on this but that never happened. I think a good point working on this is when we migrate this part of the code.